src/openms/source/FILTERING/CALIBRATION/PrecursorCorrection.cpp

```diff
@@ -201,23 +201,34 @@ namespace OpenMS
           continue;
         }
 
-        // get tolerance window and left/right iterator
+//        // get tolerance window and left/right iterator
+//        std::pair<double,double> tolerance_window = Math::getTolWindow(mz, mz_tolerance, ppm);
+//        MSSpectrum::ConstIterator left = rt_it->MZBegin(tolerance_window.first);
+//        MSSpectrum::ConstIterator right = rt_it->MZEnd(tolerance_window.second);
+//
+//        // no MS1 precursor peak in +- tolerance window found
+//        if  (left == right)
+//        {
+//          count_error_highest_intenstiy += 1;
+//          continue;
+//        }
+//
+//        MSSpectrum::ConstIterator max_intensity_it = std::max_element(left, right, Peak1D::IntensityLess());
+//
+//        // find peak (index) with highest intensity to expected position
+//        Size highest_peak_idx = max_intensity_it - rt_it->begin();
+
+        // get tolerance window and index of highest peak
         std::pair<double,double> tolerance_window = Math::getTolWindow(mz, mz_tolerance, ppm);
-        MSSpectrum::ConstIterator left = rt_it->MZBegin(tolerance_window.first);
-        MSSpectrum::ConstIterator right = rt_it->MZEnd(tolerance_window.second);
+        auto highest_peak_idx = rt_it->findHighestInWindow(mz, mz-tolerance_window.first, mz+tolerance_window.second);
```

Comment:
I would use Int here instead of auto. It's even shorter ;)

Reply:
if tests pass you can also remove the code in comments

Reply:
sure, thanks for the suggestion